libparsec/crates/protocol/tests/v3/block_create.rs

```diff
@@ -0,0 +1,97 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+// `allow-unwrap-in-test` don't behave as expected, see:
+// https://github.com/rust-lang/rust-clippy/issues/11119
+#![allow(clippy::unwrap_used)]
+
+use hex_literal::hex;
+
+use libparsec_protocol::authenticated_cmds::v3 as authenticated_cmds;
+//use libparsec_tests_fixtures::prelude::*;
+use libparsec_types::prelude::*;
+
+// Request
+
+pub fn req() {
+    // this code comes from "protocol/tests/block.rs"
+
+    // Generated from Python implementation (Parsec v2.6.0+dev)
+    // Content:
+    //   block: hex!("666f6f626172")
+    //   block_id: ext(2, hex!("57c629b69d6c4abbaf651cafa46dbc93"))
+    //   cmd: "block_create"
+    //   realm_id: ext(2, hex!("1d3353157d7d4e95ad2fdea7b3bd19c5"))
+    let raw = hex!(
+        "84a5626c6f636bc406666f6f626172a8626c6f636b5f6964d80257c629b69d6c4abbaf651c"
+        "afa46dbc93a3636d64ac626c6f636b5f637265617465a87265616c6d5f6964d8021d335315"
+        "7d7d4e95ad2fdea7b3bd19c5"
+    );
+
+    let req = authenticated_cmds::block_create::Req {
+        block_id: BlockID::from_hex("57c629b69d6c4abbaf651cafa46dbc93").unwrap(),
+        realm_id: RealmID::from_hex("1d3353157d7d4e95ad2fdea7b3bd19c5").unwrap(),
+        block: b"foobar".as_ref().into(),
+    };
+
+    let expected = authenticated_cmds::AnyCmdReq::BlockCreate(req.clone());
+    let data = authenticated_cmds::AnyCmdReq::load(&raw).unwrap();
+    assert_eq!(data, expected);
+
+    // Also test serialization round trip
+    let raw2 = req.dump().unwrap();
+    let data2 = authenticated_cmds::AnyCmdReq::load(&raw2).unwrap();
+    assert_eq!(data2, expected);
+}
+
+// Responses (required if 1 test function per status)
+
+pub fn rep_ok() {}
+pub fn rep_already_exists() {}
+pub fn rep_not_found() {}
+pub fn rep_timeout() {}
+pub fn rep_not_allowed() {}
+pub fn rep_in_maintenance() {}
+
+// Responses (required if 1 case per status)
+
+// TODO: using a raw() function to get raw values corresponding to each status. Is there a better way?
+
+pub fn raw(rep: authenticated_cmds::block_create::Rep) -> &'static [u8] {
```

Comment:
unfortunately using `authenticated_cmds::block_create::Rep` won't do the trick if the enum contains internal fields :(

Reply:
OK, I'll remove this approach